src/python/pants/option/options_bootstrapper.py

```diff
@@ -108,77 +109,81 @@ def create(
           absolute paths. Production usecases should pass True to allow options values to make the
           decision of whether to respect pantsrc files.
         """
-        env = {k: v for k, v in env.items() if k.startswith("PANTS_")}
-        args = tuple(args)
-
-        flags = set()
-        short_flags = set()
-
-        # We can't use pants.engine.fs.FileContent here because it would cause a circular dep.
-        @dataclass(frozen=True)
-        class FileContent:
-            path: str
-            content: bytes
-
-        def filecontent_for(path: str) -> FileContent:
-            return FileContent(
-                ensure_text(path),
-                read_file(path, binary_mode=True),
+        with warnings.catch_warnings(record=True):
+            env = {k: v for k, v in env.items() if k.startswith("PANTS_")}
+            args = tuple(args)
+
+            flags = set()
+            short_flags = set()
+
+            # We can't use pants.engine.fs.FileContent here because it would cause a circular dep.
+            @dataclass(frozen=True)
+            class FileContent:
+                path: str
+                content: bytes
```

Comment:
Nit. This is a lot of nesting, and it's harder to follow. Could you please move everything that does not need to be in the context manager back out to be one indentation level higher? It looks like initializing `env`, `args`, defining `file_content_for`, capture_the_flags`, and maybe `-is_bootstrap_option()`. I think only `GlobalOptions.register_bootstrap_options(capture_the_flags)` needs to be in the context manager? Maybe `Config.load_file_contents` too